Compiled from "NamingException.java"
public abstract class org.springframework.ldap.NamingException extends org.springframework.core.NestedRuntimeException {
  private java.lang.Throwable cause;

  public java.lang.Throwable getCause();
    Code:
       0: aload_0
       1: getfield      #1                  // Field cause:Ljava/lang/Throwable;
       4: aload_0
       5: if_acmpne     12
       8: aconst_null
       9: goto          16
      12: aload_0
      13: getfield      #1                  // Field cause:Ljava/lang/Throwable;
      16: areturn

  public org.springframework.ldap.NamingException(java.lang.String);
    Code:
       0: aload_0
       1: aload_1
       2: invokespecial #2                  // Method org/springframework/core/NestedRuntimeException."<init>":(Ljava/lang/String;)V
       5: return

  public org.springframework.ldap.NamingException(java.lang.String, java.lang.Throwable);
    Code:
       0: aload_0
       1: aload_1
       2: invokespecial #2                  // Method org/springframework/core/NestedRuntimeException."<init>":(Ljava/lang/String;)V
       5: aload_0
       6: aload_2
       7: putfield      #1                  // Field cause:Ljava/lang/Throwable;
      10: return

  public org.springframework.ldap.NamingException(java.lang.Throwable);
    Code:
       0: aload_0
       1: aload_1
       2: ifnull        12
       5: aload_1
       6: invokevirtual #3                  // Method java/lang/Throwable.getMessage:()Ljava/lang/String;
       9: goto          13
      12: aconst_null
      13: aload_1
      14: invokespecial #4                  // Method "<init>":(Ljava/lang/String;Ljava/lang/Throwable;)V
      17: return

  public java.lang.String getExplanation();
    Code:
       0: aload_0
       1: invokevirtual #5                  // Method getCause:()Ljava/lang/Throwable;
       4: instanceof    #6                  // class javax/naming/NamingException
       7: ifeq          21
      10: aload_0
      11: invokevirtual #5                  // Method getCause:()Ljava/lang/Throwable;
      14: checkcast     #6                  // class javax/naming/NamingException
      17: invokevirtual #7                  // Method javax/naming/NamingException.getExplanation:()Ljava/lang/String;
      20: areturn
      21: aconst_null
      22: areturn

  public javax.naming.Name getRemainingName();
    Code:
       0: aload_0
       1: invokevirtual #5                  // Method getCause:()Ljava/lang/Throwable;
       4: instanceof    #6                  // class javax/naming/NamingException
       7: ifeq          21
      10: aload_0
      11: invokevirtual #5                  // Method getCause:()Ljava/lang/Throwable;
      14: checkcast     #6                  // class javax/naming/NamingException
      17: invokevirtual #8                  // Method javax/naming/NamingException.getRemainingName:()Ljavax/naming/Name;
      20: areturn
      21: aconst_null
      22: areturn

  public javax.naming.Name getResolvedName();
    Code:
       0: aload_0
       1: invokevirtual #5                  // Method getCause:()Ljava/lang/Throwable;
       4: instanceof    #6                  // class javax/naming/NamingException
       7: ifeq          21
      10: aload_0
      11: invokevirtual #5                  // Method getCause:()Ljava/lang/Throwable;
      14: checkcast     #6                  // class javax/naming/NamingException
      17: invokevirtual #9                  // Method javax/naming/NamingException.getResolvedName:()Ljavax/naming/Name;
      20: areturn
      21: aconst_null
      22: areturn

  public java.lang.Object getResolvedObj();
    Code:
       0: aload_0
       1: invokevirtual #5                  // Method getCause:()Ljava/lang/Throwable;
       4: instanceof    #6                  // class javax/naming/NamingException
       7: ifeq          21
      10: aload_0
      11: invokevirtual #5                  // Method getCause:()Ljava/lang/Throwable;
      14: checkcast     #6                  // class javax/naming/NamingException
      17: invokevirtual #10                 // Method javax/naming/NamingException.getResolvedObj:()Ljava/lang/Object;
      20: areturn
      21: aconst_null
      22: areturn

  private void writeObject(java.io.ObjectOutputStream) throws java.io.IOException;
    Code:
       0: aload_0
       1: invokevirtual #11                 // Method getResolvedObj:()Ljava/lang/Object;
       4: astore_2
       5: aload_2
       6: instanceof    #12                 // class java/io/Serializable
       9: istore_3
      10: aload_2
      11: ifnull        60
      14: iload_3
      15: ifne          60
      18: aload_0
      19: invokevirtual #5                  // Method getCause:()Ljava/lang/Throwable;
      22: checkcast     #6                  // class javax/naming/NamingException
      25: astore        4
      27: aload         4
      29: aconst_null
      30: invokevirtual #13                 // Method javax/naming/NamingException.setResolvedObj:(Ljava/lang/Object;)V
      33: aload_1
      34: invokevirtual #14                 // Method java/io/ObjectOutputStream.defaultWriteObject:()V
      37: aload         4
      39: aload_2
      40: invokevirtual #13                 // Method javax/naming/NamingException.setResolvedObj:(Ljava/lang/Object;)V
      43: goto          57
      46: astore        5
      48: aload         4
      50: aload_2
      51: invokevirtual #13                 // Method javax/naming/NamingException.setResolvedObj:(Ljava/lang/Object;)V
      54: aload         5
      56: athrow
      57: goto          64
      60: aload_1
      61: invokevirtual #14                 // Method java/io/ObjectOutputStream.defaultWriteObject:()V
      64: return
    Exception table:
       from    to  target type
          33    37    46   any
          46    48    46   any
}
